### PR TITLE
[TASK] Add best practices for logging

### PR DESCRIPTION
**Pull request by DanielSiepmann**
As these are not strict rules but only best practices,
they still should be available to everyone.
This would make it possible to have the streamlined result throughout
the whole ecosystem.

### CODE REVIEW

**Comment by sypets** [comment]
Do you want to add 1 or 2 examples? Maybe a simple example and one for cases that are not so entirely obvious, e.g. information about exception.

File: Documentation/ApiOverview/Logging/Logger/Index.rst [hunk 1]
```diff
@@ -119,3 +119,37 @@ For each of the severity levels mentioned above, a shorthand method exists in
...
 - :code:`$this->logger->notice($message, array $data = array());`
 - etc.
+
+.. _logging-logger-best-practices:
+
+Best Practices
+==============
+
+There are no strict rules or guidelines about logging.
+But it can be considered to be best practice to follow these rules:
```
Comment (oliverklee) [inline Documentation/ApiOverview/Logging/Logger/Index.rst]: But -> Still,

File: Documentation/ApiOverview/Logging/Logger/Index.rst [hunk 2]
```diff
@@ -119,3 +119,37 @@ For each of the severity levels mentioned above, a shorthand method exists in
...
+
+Meaningful message
+   The message itself has to be meaningful, like Exception messages.
```
Comment (oliverklee) [inline Documentation/ApiOverview/Logging/Logger/Index.rst]: like -> for example

File: Documentation/ApiOverview/Logging/Logger/Index.rst [hunk 3]
```diff
@@ -119,3 +119,37 @@ For each of the severity levels mentioned above, a shorthand method exists in
...
+
+Searchable message
+   Most of the times log entries will be stored.
+   They are most important if something goes wrong within the system.
+   In such situations people might search for specific issues or situations.
```
Comment (oliverklee) [inline Documentation/ApiOverview/Logging/Logger/Index.rst]: Add comma after "situations"

**Comment by oliverklee** [comment]
Good work! I have added some comments.

**Comment by oliverklee** [comment]
Concerning examples, I really love them to be scannable (i.e., with a heading and one example per line), and there to be multiple ones. Like this:

Bad:
`foo`
`bar`
`baz`

Good:
`frubble`
`gribble`
`robble`

**Pull request merged**